modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -1141,6 +1098,42 @@ def _apply_func_to_list_of_partitions(cls, func, partitions, **kwargs):
         preprocessed_func = cls.preprocess_func(func)
         return [obj.apply(preprocessed_func, **kwargs) for obj in partitions]
 
+    @classmethod
+    def combine(cls, partitions):
+        """
+        Convert a NumPy 2D array of partitions to a NumPy 2D array of a single partition.
+
+        Parameters
+        ----------
+        partitions : np.ndarray
+            The partitions which have to be converted to a single partition.
+
+        Returns
+        -------
+        np.ndarray
+            A NumPy 2D array of a single partition.
+        """
+
+        def to_pandas_remote(data, partition_shape, *partition_data):
```

Comment:
`data` parameter is not used anywhere. Should be removed.

Reply:
data parameter  is by default present in any function passed to partition.apply() this would hold the data of partition, I am quite not sure if it would be possible to remove this.

Reply:
Ah, I see. Then we should probably pass partition_refs[1:] in this function and combine it with data argument to avoid an extra deserialization of a pandas dataframe.

Reply:
This is smart. Thanks